packages/components/src/components/post-pagination/post-pagination.tsx

```diff
@@ -0,0 +1,844 @@
+    return secondRect.left - firstRect.right;
+  }
+
+  /**
+   * Measures actual rendered elements to determine how many pages can fit
+   */
+  private measureAndCalculateVisiblePages() {
+    if (!this.navRef || !this.hiddenItemsRef || !this.connected) {
+      return;
+    }
+
+    const totalPages = this.getTotalPages();
+    if (totalPages <= 1) {
+      return;
+    }
+
+    const paginationPadding = this.getPaginationPadding();
+    const availableWidth = this.getAvailableWidth();
+    const netWidth = availableWidth - paginationPadding;
+
+    const controlButtonsWidth = this.getControlButtonsWidth();
+    const pageButton = this.hiddenItemsRef.querySelector('.hidden-page-button');
+    const ellipsis = this.hiddenItemsRef.querySelector('.hidden-ellipsis');
+    
+    if (!pageButton) {
+      return;
+    }
+
+    const singleButtonWidth = pageButton.getBoundingClientRect().width;
+    const gap = pageButton && ellipsis ? this.calculateGap(pageButton, ellipsis) : 0;
+    const widthForPages = netWidth - controlButtonsWidth;
+
+    // Calculate how many page buttons can fit
+    const maxPages = Math.floor((widthForPages + gap) / (singleButtonWidth + gap));
+    const clampedMaxPages = Math.max(MIN_VISIBLE_PAGES, Math.min(maxPages, totalPages));
+    
+    this.maxVisiblePages = clampedMaxPages;
+    this.updatePagesWithValidation();
+  }
+
+  /**
+   * Gets total width of control buttons (prev/next)
+   */
+  private getControlButtonsWidth(): number {
+    if (!this.hiddenItemsRef) return 0;
+    
+    const controlButtons = Array.from(
+      this.hiddenItemsRef.querySelectorAll('.hidden-control-button')
+    );
+    
+    const totalWidth = controlButtons.reduce((sum, el) => {
+      return sum + el.getBoundingClientRect().width;
+    }, 0);
+    
+    return totalWidth;
+  }
+
+  /** 
+   * Gets the horizontal padding of the pagination container
+   */
+  private getPaginationPadding(): number {
+    if (!this.navRef) return 0;
+    const computedStyle = window.getComputedStyle(this.navRef);
+    const paddingLeft = parseFloat(computedStyle.paddingLeft) || 0;
+    const paddingRight = parseFloat(computedStyle.paddingRight) || 0;
+    return paddingLeft + paddingRight;
+  }
+
+  /**
+   * Gets available width from parent container
+   */
+  private getAvailableWidth(): number {
+    if (!this.navRef) return 0;
+    
+    let parent = this.host.parentNode;
+    while (parent && !(parent instanceof HTMLElement)) {
+      parent = parent.parentNode;
+    }
+    
+    return parent instanceof HTMLElement ? parent.clientWidth : window.innerWidth;
+  }
+
+  /**
+   * Clamps the page number to valid range
+   */
+  private clampPageToValidRange(totalPages: number): number {
+    if (totalPages === 0 || this.collectionSize === 0 || isNaN(totalPages)) {
+      return 1;
+    }
+    if (!this.page || this.page < 1 || isNaN(this.page)) {
+      return 1;
+    }
+    if (this.page > totalPages) {
+      return totalPages;
+    }
```

Comment:
Can you create a constant per if statement above with appropriate names (so other DEVs can understand which statement means what) and then put the if statemens in an if-else structure, for readability?

Also the first and second statements return the same value. Can we combine them?

e.g. `if(invalidTotalPages || invalidPage) { ... }`